Automattic/simplenote-macos
# Preferences View Controller and UI

### Fix
This is the first PR in the process of adding the Preferences window to Simplenote Mac (over all issue #1037, fixes #1038 )

In this PR I have added all of the UI for the preferences view and linked all of the controls into the window.

<img height=400 src="https://cdn-std.droplr.net/files/acc_593859/KCRls9"/>

Note: one thing that is not done here is the button color for the Delete Account  button.  Still working on getting that going.

### Test
1. Test opening the preferences window by going to Simplenote > Preferences, hitting Command + , and clicking on the Settings button in the tag list
2. Make sure that you can only open one preferences view at a time (shouldn't be possible to have multiple windows of preferences appear)
3. Check and see if your account email appears in the top
4. Test log out button
5. log in with a testing account, test delete account button
6. change the sort order of your notes and make sure the note list updates
7. change the line length setting from full to narrow and back
8. select the condensed note list and deselect it
9. select the sort tags alphabetically and deselect it
10. try the different theme settings
11. use the font slider to change the font size
12. test to make sure these setting changes persist when you close the preferences window and when you restart the app
13. click on the "About Analytics and Privacy" link to be taken to the a8c privacy page

### Review
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release

> `RELEASE-NOTES.txt` was updated in a4948377 with:
> 
> > Added preferences menu to centralize app options and customizations #1042
